File: tests/expected/cast.jl

function cast_types()
    a = Int64(floor(float(1)))
    println(join([a], " "))
end

function cast_ctypes()
    a = convert(Int16, 1)
    b = a
    println(join([b], " "))
    c = convert(Int64, 1)
    d = c
    println(join([d], " "))
    e = convert(UInt64, 1)
    f = e
    println(join([f], " "))
end

function main()
    cast_types()
    cast_ctypes()
end

main()

File: tests/expected/cast.go
package main

import (
	"fmt"
)

func CastTypes() {
	var a int = int(float64(1))
	fmt.Printf("%v\n", a)
}

func CastCtypes() {
	var a int16 = int16(1)
	b := a
	fmt.Printf("%v\n", b)
	var c int64 = int64(1)
	d := c
	fmt.Printf("%v\n", d)
	var e uint64 = uint64(1)
	f := e
	fmt.Printf("%v\n", f)
}

func main() {
	CastTypes()
	CastCtypes()
}
